src/pages/CountrySelectorPage.js

```diff
@@ -0,0 +1,98 @@
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import React, {useCallback, useMemo, useState} from 'react';
+import PropTypes from 'prop-types';
+import withLocalize, {withLocalizePropTypes} from '../components/withLocalize';
+import withCurrentUserPersonalDetails from '../components/withCurrentUserPersonalDetails';
+import ScreenWrapper from '../components/ScreenWrapper';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import Navigation from '../libs/Navigation/Navigation';
+import * as Expensicons from '../components/Icon/Expensicons';
+import themeColors from '../styles/themes/default';
+import OptionsSelector from '../components/OptionsSelector';
+import styles from '../styles/styles';
+import compose from '../libs/compose';
+
+const propTypes = {
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            backTo: PropTypes.string,
+        }),
+    }).isRequired,
+
+    /** User's private personal details */
+    currentUserPersonalDetails: PropTypes.shape({
+        /** User's home address */
+        address: PropTypes.shape({
+            country: PropTypes.string,
+        }),
+    }).isRequired,
+
+    ...withLocalizePropTypes,
+};
+
+const greenCheckmark = {src: Expensicons.Checkmark, color: themeColors.success};
+
+function filterCountries(searchValue, countries) {
+    const trimmedSearchValue = searchValue.trim();
+    if (trimmedSearchValue.length === 0) {
+        return [];
+    }
+
+    return _.filter(countries, country => country.text.toLowerCase().includes(searchValue.toLowerCase()));
+}
+
+function CountrySelectorPage(props) {
```

Comment:
DRY, This file has a lot of similarities with the `StateSelectorPage` component. Could you consider refactoring this code to create more reusable components that can be shared across multiple files?